docs/privacy/cspp.md

```diff
@@ -0,0 +1,79 @@
+# CoinShuffle++
+
+To enhance privacy, Decred has implemented a variant of CoinShuffle++ in its wallet.
+
+---
+
+## What is CoinShuffle++ and how does it work?
+
+**CoinShuffle++ (CSPP)** is a mixing protocol used to create Decred CoinJoin transactions.
+
+CSPP is a **non-custodial** process, where the output addresses are **anonymized** via a **mixnet**. When using the CSPP mixnet, note the CoinJoin leaks which inputs and change addresses belong to which peer to the server, but not the other peers. The outputs are fully anonymized, none of the peers or the server can link outputs and inputs.
+
+Decred's implementation is based on the CoinShuffle++ protocol from ["P2P Mixing and Unlinkable Bitcoin Transactions"](https://eprint.iacr.org/2016/824.pdf) by Ruffing, Moreno-Sanchez and Kate. It uses [**DiceMix Light**](https://github.com/ElementsProject/dicemix/blob/master/doc/protocol.md), a faster iteration by Ruffing on the DiceMix process proposed in the CoinShuffle++ paper. This process allows to create untraceable transactions, but the amounts are still publicly available.  To make the outputs indistinguishable, they must have a fixed denomination for each mix.
+
+CoinShuffle++ does a fine job of anonymizing the output addresses, but if the change is not handled with care, it can link mixed and unmixed UTXOs. In many cases, change outputs can be linked to their inputs by doing a partial sum analysis. To deal with this threat, **change from mixes flows to a separate wallet account**, where it is then mixed into smaller denominations until the change is less than the smallest mixer denomination.
+
+Mixes occur episodically in **epochs**, with the mainnet epoch set to **20 minutes** (1200 seconds).
+
+---
+
+## How do I use CSPP?
+
+To use CSPP, several new options must be set in `dcrwallet.conf`:
+
+    - csppserver=cspp.decred.org:15760 - This is the FQDN and port csppserver is listening on.
+    - csppserver.ca=cspp.decred.org.pem - This is the CA certificate for the csppserver.
+    - purchaseaccount=mixed - This sets the account from which you will buy tickets.
+    - mixaccount=mixed/1 - As mixes occur, mixed output addresses will come from the internal branch this account.
+    - changeaccount=unmixed - Each set of mix inputs uses a change address from this account.
+    - mixchange=1 - This tells dcrwallet to mix payments to changeaccount into the mixedaccount.
+    - ticketbuyer.votingaccount=voting - This tells the wallet which account to use when setting voting ticket voting addresses.
+
+In addition to setting these new options in `dcrwallet.conf`, **two new accounts must be created: mixed and unmixed.** The voting account can either be created locally, if your `ticketbuyer` wallet also votes, or you can import an extended pubkey for a voting account from an external voting wallet.
```

Comment:
Could use some more detail on how to export and import xpub

Reply:
I'll add the cspp.decred.org and detailed info about exporting and importing xpubs in the two sections mentioned above.

Reply:
The initial reasoning for this page was so the cspp daemon didn't need to run http.

Reply:
RIght, so this page should reproduce all of the useful practical info on cspp.decred.org then.